elasticdl/python/data/reader/odps_reader.py

```diff
@@ -29,10 +29,10 @@ def _init_metadata(self):
             )
 
         if self._metadata.column_names:
-            self._metadata.column_dtypes = [
-                table_schema[column_name].type
-                for column_name in self._metadata.column_names
-            ]
+            column_dtypes = {}
+            for column_name in self._metadata.column_names:
```

Comment:
column_dtypes = {column_name: table_schema[column_name].type for column_name in self._metadata.column_names}